tests/gtests/api/test_memory_desc.cpp

```diff
@@ -88,4 +88,54 @@ HANDLE_EXCEPTIONS_FOR_TEST(memory_desc_test_t, TestComparison) {
     EXPECT_NE(plain_md, blocked_md);
 }
 
+TEST(c_api_host_scalar_md, TestHostScalarF32) {
+    dnnl_memory_desc_t scalar_md = nullptr;
+    DNNL_CHECK(dnnl_memory_desc_create_host_scalar(&scalar_md, dnnl_f32));
+
+    int ndims = -1;
+    dnnl_data_type_t dtype = dnnl_data_type_undef;
+    dnnl_format_kind_t fmt_kind = dnnl_format_kind_undef;
+
+    DNNL_CHECK(dnnl_memory_desc_query(scalar_md, dnnl_query_ndims_s32, &ndims));
+    EXPECT_EQ(ndims, 1);
+
+    DNNL_CHECK(dnnl_memory_desc_query(scalar_md, dnnl_query_data_type, &dtype));
+    EXPECT_EQ(dtype, dnnl_f32);
+
+    DNNL_CHECK(dnnl_memory_desc_query(
+            scalar_md, dnnl_query_format_kind, &fmt_kind));
+    EXPECT_EQ(fmt_kind, dnnl_format_kind_host_scalar);
+
+    size_t sz = dnnl_memory_desc_get_size(scalar_md);
+    EXPECT_EQ(sz, sizeof(float));
+
+    DNNL_CHECK(dnnl_memory_desc_destroy(scalar_md));
+}
+
+TEST(cpp_api_host_scalar_md, TestHostScalarF32) {
+    using namespace dnnl;
+    auto md = memory::desc::host_scalar(memory::data_type::f32);
+
+    EXPECT_EQ(md.get_ndims(), 1);
+    EXPECT_EQ(md.get_data_type(), memory::data_type::f32);
+    EXPECT_EQ(md.get_format_kind(), memory::format_kind::host_scalar);
+    EXPECT_EQ(md.get_size(), sizeof(float));
+}
+
+TEST(c_api_host_scalar_md, TestHostScalarNullPtr) {
+    dnnl_status_t status
+            = dnnl_memory_desc_create_host_scalar(nullptr, dnnl_f32);
+    ASSERT_EQ(status, dnnl_invalid_arguments);
+}
+
+TEST(c_api_host_scalar_md, TestHostScalarInvalidType) {
+    dnnl_memory_desc_t scalar_md {};
+
+    dnnl_status_t status = dnnl_memory_desc_create_host_scalar(
+            &scalar_md, (dnnl_data_type_t)999);
+    ASSERT_EQ(status, dnnl_invalid_arguments);
+
+    DNNL_CHECK(dnnl_memory_desc_destroy(scalar_md));
+}
```

Comment:
Better check md with `data_type::undef` instead.

Reply:
I think that would be different kind of testing.
Right now the idea is to input an "invalid" (or garbage in some sense?) value and see what happens. What you suggest is to check what function is supposed to be supporting I guess. I'm OK with both approaches, so I will add an additional tests case to this test.

Reply:
Updated test cases